### PR TITLE
Weapon to Component Transition

### PR DESCRIPTION
**Pull request by UnoYakshi**
Moved weapon "interface" to component (from `ProtCharacter`).
Moved @andrew-it's aiming from BP to code.
Fixed left hand's IK to a more stable version via BoneSpace (depends on the right hand).

### CODE REVIEW

File: Source/Prot/ProtCharacter.h [hunk 1]
```diff
@@ -5,11 +5,11 @@
 #include "CoreMinimal.h"
 #include "MyPlayerController.h"
 #include "Weapons/Weapon.h"
```
Comment (UnoYakshi) [inline Source/Prot/ProtCharacter.h:7]: Remove `Weapons/Weapon.h`.

File: Source/Prot/Weapons/EquipInterface.h [hunk 2]
```diff
@@ -0,0 +1,25 @@
+// Fill out your copyright notice in the Description page of Project Settings.
```
Comment (UnoYakshi) [inline Source/Prot/Weapons/EquipInterface.h]: Remove the interface.

File: Source/Prot/Weapons/FireInterface.h [hunk 3]
```diff
@@ -0,0 +1,43 @@
+// Fill out your copyright notice in the Description page of Project Settings.
```
Comment (UnoYakshi) [inline Source/Prot/Weapons/FireInterface.h]: Remove the interface.

File: Source/Prot/Weapons/InteractiveInterface.h [hunk 4]
```diff
@@ -0,0 +1,25 @@
+// Fill out your copyright notice in the Description page of Project Settings.
```
Comment (UnoYakshi) [inline Source/Prot/Weapons/InteractiveInterface.h]: Remove the interface.

File: Source/Prot/Weapons/WeaponComponent.cpp [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+		CurrentWeapon->SetActorRelativeLocation(FVector::ZeroVector);
+		CurrentWeapon->OnEquip(nullptr);
+	}
+}
+
+void UWeaponComponent::OnComponentDestroyed(bool bDestroyingHierarchy)
+{
+	Super::OnComponentDestroyed(bDestroyingHierarchy);
+
+	//StopSimulatingWeaponFire();
```
Comment (UnoYakshi) [inline Source/Prot/Weapons/WeaponComponent.cpp]: Uncomment.

File: Source/Prot/ProtCharacter.cpp [hunk 6]
```diff
@@ -618,13 +440,33 @@ void AProtCharacter::GetLifetimeReplicatedProps(TArray< FLifetimeProperty > & Ou
...
+	// Get direction vectors from Character and PC...
+	FVector ActorDir = GetActorRotation().Vector();
+	FVector PCDir = FRotator(0.f, Input, 0.f).Vector();
+
+	// Compute the Angle difference between the two direction
+	float Angle = FMath::Acos(FVector::DotProduct(ActorDir, PCDir));
+	Angle = FMath::RadiansToDegrees(Angle);
+
+	// Find on which side is the angle difference (left or right)
+	FRotator Temp = GetActorRotation() - FRotator(0.f, 90.f, 0.f);
```
Comment (andrew-it) [inline Source/Prot/ProtCharacter.cpp:460]: What is it?

**Review by andrew-it — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**